### PR TITLE
add decrease functionality

### PR DESCRIPTION
**Pull request by Mati0x**
#160

User should be able to unStake tokens from community if member has previously registered and added stake

A toast notification with error name is open when user try to unStake with no prior added staked or user try to unStake more than added stake

Add "addedStake" info to user = every added stake member has done in community that is NOT from registering in community
Also add some disabled states for guide the user: disable if: member try to decrease more than staked

Screenshot 2024-05-06 at 18 20 58

### CODE REVIEW

File: apps/web/components/IncreasePower.tsx [hunk 1]
```diff
@@ -163,6 +164,33 @@ export const IncreasePower = ({
...
+    useTransactionNotification(decreasePowerData);
+
+  useEffect(() => {
+    updateDecreasePowerTransactionStatus(decreasePowerStatus);
+  }, [decreasePowerStatus]);
+
+  const decreasePoweErrorName = useErrorDetails(errorDecreasePower);
+  useEffect(() => {
+    if (isErrordecreasePower && decreasePoweErrorName.errorName !== undefined) {
+      alert("UNEXECUTABLE OPTION:" + " " + decreasePoweErrorName.errorName);
```
Comment (kamikazebr) [inline apps/web/components/IncreasePower.tsx:191]: use toast

**Comment by kamikazebr** [comment]
Still missing the current value that user has to know how much they can decrease, buts the button its working.

File: apps/web/components/IncreasePower.tsx [hunk 2]
```diff
@@ -163,6 +165,33 @@ export const IncreasePower = ({
...
+
+  useEffect(() => {
+    updateDecreasePowerTransactionStatus(decreasePowerStatus);
+  }, [decreasePowerStatus]);
+
+  const decreasePoweErrorName = useErrorDetails(errorDecreasePower);
+  useEffect(() => {
+    if (isErrordecreasePower && decreasePoweErrorName.errorName !== undefined) {
+      toast.error(decreasePoweErrorName.errorName);
+    }
```
Comment (kamikazebr) [inline apps/web/components/IncreasePower.tsx:193]: if some error occours but that dont have errorName that its possible? so we dont have notification?

Comment (pglavin2) [reply apps/web/components/IncreasePower.tsx:193]: looks like there is a typo in "decreasePoweErrorName" - missing "r" in Power

**Comment by Mati0x** [comment]
> Button is working for me and looks OK to merge. Here's some more improvements to work on soon:
> 
> * display current tokens stake (registration stake + increasePower)
> * only display "Decrease Stake" if there are already additional tokens staked
> * only display the entire "Customize your stake in the community" section if you are already registered in the community

Yess I have a much better improvement to make for Beta for this section, and have considered those thing you suggested. Momentary for alpha I just added the functionality to test if works

File: apps/web/components/IncreasePower.tsx (hunk 2 above)
Comment (kamikazebr) [reply apps/web/components/IncreasePower.tsx:193]: @Mati0x missing that to merge

Comment (kamikazebr) [reply apps/web/components/IncreasePower.tsx:193]: @Mati0x just fix the typo for now and we are good to merge

**Comment by kamikazebr** [comment]
Need be checked in sepolia after #154

**Review by kamikazebr — APPROVED** [review]

**Pull request merged**